shadow_cache.c

```diff
@@ -64,6 +64,9 @@ int shadow_cache_get(const char *name, char **entry TSRMLS_DC)
 	namelen = shadow_cache_segmented_name(&segname, name TSRMLS_CC);
 	zend_string *segname_zs = zend_string_init(segname, namelen, 0);
 	if ((centry = zend_hash_find(&SHADOW_G(cache), segname_zs)) != NULL) {
+        if(Z_STRLEN_P(centry) == 0){
```

Comment:
If we have an empty cache entry we should return Null as that's the expected return value for emptys